lib/exercises.rb

```diff
@@ -1,19 +1,44 @@
 
 # This method will return an array of arrays.
 # Each subarray will have strings which are anagrams of each other
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: o(n)
+# Space Complexity: o(n)
 
 def grouped_anagrams(strings)
-  raise NotImplementedError, "Method hasn't been implemented yet!"
+  return [] if strings.length == 0
+
+  anagrams = Hash.new([])
+  strings.each do |element|
+    sorted_letters = element.split('').sort
+    if anagrams[sorted_letters]
+      anagrams << element
```

Comment:
Note here `anagrams` is a Hash and not an array, to get the array you need `anagrams[sorted_letters]

```suggestion
      anagrams[sorted_letters] << element
```